File: go/x1_watch.go
package main

import (
    . "g2d"
    "math"
)

func main() {
    radius1, radius2, radius3 := 300.0, 295.0, 280.0
    x, y := radius1, radius1
    InitCanvas(Point{int(radius1*2), int(radius1*2)})
    SetColor(Color{0, 0, 0})
    for i := 0; i < 12; i++ {
        angle := float64(i) * 2 * math.Pi / 12
        pt1 := Point{int(x + radius1*math.Cos(angle)), int(y + radius1*math.Sin(angle))}
        pt2 := Point{int(x + radius3*math.Cos(angle)), int(y + radius3*math.Sin(angle))}
        DrawLine(pt1, pt2)
    }
    for i := 0; i < 60; i++ {
        angle := float64(i) * 2 * math.Pi / 60
        pt1 := Point{int(x + radius1*math.Cos(angle)), int(y + radius1*math.Sin(angle))}
        pt2 := Point{int(x + radius2*math.Cos(angle)), int(y + radius2*math.Sin(angle))}
        DrawLine(pt1, pt2)
    }
    MainLoop()
}

File: examples/x1_watch.py
#!/usr/bin/env python3
'''
@author  Michele Tomaiuolo - http://www.ce.unipr.it/people/tomamic
@license This software is free - http://www.gnu.org/licenses/gpl.html
'''

import g2d
from math import pi, sin, cos

R = 200

g2d.init_canvas((2*R, 2*R))
g2d.set_color((0, 0, 0))

for i in range(60):
    r = R - 5
    if i % 5 == 0:
        r = R - 20
    angle = i * (2*pi / 60)
    pt1 = R + int(r*cos(angle)), R + int(r*sin(angle))
    pt2 = R + int(R*cos(angle)), R + int(R*sin(angle))
    g2d.draw_line(pt1, pt2)

g2d.main_loop()
